packages/babel-runtime-corejs3/helpers/esm/wrapRegExp.js

```diff
@@ -42,7 +42,7 @@ function _wrapRegExp() {
   }, BabelRegExp.prototype[_Symbol$replace] = function (t, p) {
     if ("string" == typeof p) {
       var o = r.get(this);
-      return e[_Symbol$replace].call(this, t, p.replace(/\$<([^>]+)>/g, function (e, r) {
+      return e[_Symbol$replace].call(this, t, p.replace(/\$<(?!\$<)([^>]+)>/g, function (e, r) {
```

Comment:
This doesn't actually solve the problem, since you could have a string that looks like `$<a$<a$<a$<a$<a$<a$<a$<a$<a$<a$<a$<a...` and this regexp wouldn't protect from it.